pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -379,88 +388,463 @@ func Test_server_IndexInfo(t *testing.T) {
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want, *payload.Info_Index_Count, error) error
-		beforeFunc func(args)
+		checkFunc  func(*server, context.Context, args, want, *payload.Info_Index_Count, error) error
+		beforeFunc func(*testing.T, context.Context, args, *server)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, gotRes *payload.Info_Index_Count, err error) error {
+
+	// common variables for test
+	const (
+		name      = "vald-agent-ngt-1" // agent name
+		dim       = 3                  //  vector dimension
+		id        = "uuid-1"           // id for getObject request
+		insertCnt = 100                // default insert count
+		removeCnt = 100                // default remove count
+	)
+	var (
+		ip = net.LoadLocalIP() // agent ip address
+
+		// default NGT configuration for test
+		kvsdbCfg  = &config.KVSDB{}
+		vqueueCfg = &config.VQueue{}
+
+		defaultSvcCfg = &config.NGT{
+			Dimension:    dim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Float.String(),
+			KVSDB:        kvsdbCfg,
+			VQueue:       vqueueCfg,
+		}
+		defaultSvcOpts = []service.Option{
+			service.WithEnableInMemoryMode(true),
+		}
+
+		defaultInsertConfig = &payload.Insert_Config{}
+		defaultRemoveConfig = &payload.Remove_Config{}
+	)
+	genReqAndInsert := func(s *server, ctx context.Context, cnt int, createIdx bool) (*payload.Insert_MultiRequest, error) {
```

Comment:
```suggestion
	beforeInsertAndCreateIndex := func(s *server, ctx context.Context, cnt int, createIdx bool) (*payload.Insert_MultiRequest, error) {
```

Reply:
you have to rename it.
this function has much processing and it's hard to understand what to do from the function name.

Reply:
Which part do you think is hard to understand?

In my opinion this function only generates random vector and insert it into the server, and create index if necessary.

Reply:
the implementation seems complex but most of them are error handling, I don't think it is complex.

Reply:
sorry i misread your comment, i will fix it

Reply:
fixed